### PR TITLE
[OF-1420] refac: Add ReferenceID to Agora token

### PR DESCRIPTION
**Pull request by MAG-DanielMurdolo**
connected-spaces-platform Pull Request

**For the reviewer**

* [x] If required, are the changes covered by appropriate tests?
* [x] Are any public-facing API changes well documented?
* [x] Is the code easily readable and extensible and/or follow existing conventions?

### CODE REVIEW

File: Library/include/CSP/Systems/Users/Authentication.h [hunk 1]
```diff
@@ -149,11 +149,12 @@ class CSP_API LoginTokenInfoResult : public ResultBase
 };
 
 
-/// @brief Data structure for an Agora user token, giving userID, channel name and settings regarding sharing of audio/video/screenshare.
+/// @brief Data structure for an Agora user token, giving userID, referenceID, channel name and settings regarding sharing of audio/video/screenshare.
 class CSP_API AgoraUserTokenParams
 {
 public:
 	csp::common::String AgoraUserId;
```
Comment (MAG-SamBirley) [inline Library/include/CSP/Systems/Users/Authentication.h:156]: I don't actually know what a `ReferenceID` does in this context, could you add a doxygen-style comment explaining it's purpose?

Bonus points for adding similar comments to other member variables whilst you're in the area! 😁

**Review by MAG-SamBirley — CHANGES_REQUESTED** [review]

**Review by MAG-mv — CHANGES_REQUESTED** [review]

File: Library/include/CSP/Systems/Users/Authentication.h (hunk 1 above)
Comment (MAG-DanielMurdolo) [reply Library/include/CSP/Systems/Users/Authentication.h:157]: Added Doxygen comments for each field

File: Library/include/CSP/Systems/Users/Authentication.h [hunk 2]
```diff
@@ -149,11 +149,12 @@ class CSP_API LoginTokenInfoResult : public ResultBase
 };
 
 
-/// @brief Data structure for an Agora user token, giving userID, channel name and settings regarding sharing of audio/video/screenshare.
+/// @brief Data structure for an Agora user token, giving userID, referenceID, channel name and settings regarding sharing of audio/video/screenshare.
```
[the remark this answers is not in the document]
Comment (MAG-DanielMurdolo) [reply Library/include/CSP/Systems/Users/Authentication.h:152]: Added Doxygen comments for each field

**Review by MAG-SamBirley — DISMISSED** [review]
Looks great, thanks!

**Review by MAG-mv — APPROVED** [review]

**Review by MAG-SamBirley — APPROVED** [review]

**Pull request merged**